src/clj/game/cards/resources.clj

```diff
@@ -1973,6 +1974,18 @@
     :leave-play (req (remove-watch state :theophilius-bagbiter)
                      (swap! state assoc-in [:runner :hand-size :base] 5))}
 
+   "Thunder Art Gallery"
+   (let [ability {:once :per-turn
+                  :optional {:prompt (msg "Use Thunder Art Gallery to install a card?")
+                             :yes-ability {:choices (req (cancellable (remove #(is-type? % "Event") (:hand runner))))
+                                           :prompt (msg "Which card to install?")
+                                           :effect (req (install-cost-bonus state side [:credit -1])
+                                                        (when (runner-can-install? state side target)
+                                                              (system-msg state side "uses Thunder Art Gallery to install a card.")
+                                                              (runner-install state side target)))}}}]
```

Comment:
`runner-install` is an async operation, so the ability itself should be `:async true` and pass its `eid` to `runner-install`. `effect-completed` will then need to be called both in the false branch of the conditional above (converting to `if`) and in the `:cancel-effect` that we'll have to add.